oak/common/nonce_generator.cc

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak/common/nonce_generator.h"
+
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+#include "absl/strings/escaping.h"
+
+namespace oak {
+
+namespace {
+
+// Copied from https://github.com/google/asylo/blob/master/asylo/crypto/util/bssl_util.cc.
+std::string BsslLastErrorString() {
+  static constexpr int kErrorStringBufferLength = 256;
+  static char buffer[kErrorStringBufferLength];
+  ERR_error_string_n(ERR_get_error(), buffer, sizeof(buffer));
+  ERR_clear_error();
+  return std::string(buffer);
+}
+
+}  // namespace
+
+NonceGenerator::NonceGenerator(size_t nonce_size_bytes) : nonce_size_bytes_(nonce_size_bytes) {}
+
+// Copied from https://github.com/google/asylo/blob/master/asylo/crypto/random_nonce_generator.cc.
+asylo::StatusOr<Nonce> NonceGenerator::NextNonce() {
```

Comment:
Don't we have optional? I don't like carrying asylo everywhere :)

Reply:
This is not the entire asylo, it's just their status stuff, which I think is perfectly fine to pull in, as opposed to reimplement it ourselves or use sub par alternatives.

Reply:
Anyways, using `std::random_device` I don't need this any more :) but I still think we should use it where it makes sense.